docs/topics/basic-types.md

```diff
@@ -3,613 +3,5 @@
-//sampleEnd
-}
-```
-{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
-
-You can use templates both in raw and escaped strings.
-To insert the `$` character in a raw string (which doesn't support backslash escaping) before any symbol, which is allowed as a beginning of an [identifier](https://kotlinlang.org/docs/reference/grammar.html#identifiers), use the following syntax:
-
-```kotlin
-val price = """
-${'$'}_9.99
-"""
-```
-
-## Arrays
-
-Arrays in Kotlin are represented by the `Array` class. It has `get` and `set` functions that turn into `[]` by operator overloading conventions,
- and the `size` property, along with other useful member functions:
-
-```kotlin
-class Array<T> private constructor() {
-    val size: Int
-    operator fun get(index: Int): T
-    operator fun set(index: Int, value: T): Unit
-
-    operator fun iterator(): Iterator<T>
-    // ...
-}
-```
-
-To create an array, use the function `arrayOf()` and pass the item values to it, so that `arrayOf(1, 2, 3)` creates an array `[1, 2, 3]`.
-Alternatively, the `arrayOfNulls()` function can be used to create an array of a given size filled with `null` elements.
-
-Another option is to use the `Array` constructor that takes the array size and the function that returns values
-of array elements given its index:
-
-```kotlin
-fun main() {
-//sampleStart
-    // Creates an Array<String> with values ["0", "1", "4", "9", "16"]
-    val asc = Array(5) { i -> (i * i).toString() }
-    asc.forEach { println(it) }
-//sampleEnd
-}
-```
-{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
-
-As we said above, the `[]` operation stands for calls to member functions `get()` and `set()`.
-
-Arrays in Kotlin are _invariant_. This means that Kotlin does not let us assign an `Array<String>`
-to an `Array<Any>`, which prevents a possible runtime failure (but you can use `Array<out Any>`, 
-see [Type Projections](generics.md#type-projections)).
-
-### Primitive type arrays
-
-Kotlin also has classes that represent arrays of primitive types without boxing overhead: `ByteArray`,
-`ShortArray`, `IntArray`, and so on. These classes have no inheritance relation to the `Array` class, but they
-have the same set of methods and properties. Each of them also has a corresponding factory function:
-
-```kotlin
-val x: IntArray = intArrayOf(1, 2, 3)
-x[0] = x[1] + x[2]
-```
-
-```kotlin
-// Array of int of size 5 with values [0, 0, 0, 0, 0]
-val arr = IntArray(5)
-
-// e.g. initialise the values in the array with a constant
-// Array of int of size 5 with values [42, 42, 42, 42, 42]
-val arr = IntArray(5) { 42 }
-
-// e.g. initialise the values in the array using a lambda
-// Array of int of size 5 with values [0, 1, 2, 3, 4] (values initialised to their index value)
-var arr = IntArray(5) { it * 1 } 
-```
+In this section we describe the basic types used in Kotlin: [numbers](numbers.md), [booleans](booleans.md), 
+[characters](characters.md), [strings](strings.md), and [arrays](arrays.md).
```

Comment:
Although, they include advanced concepts such as arrays and ranges, so maybe it's ok to omit them from basic types list here. Those who are curious, could get there from ToC.

Reply:
I think it's better to add rather than to omit. We decided to place it inside the "Basic types" section, so why not include it here too. Thanks, [added](https://github.com/JetBrains/kotlin-web-site/pull/2409/commits/992bb5faa756409f778f1369c2fe0bd9eed4054b).